wgrib 3.5.0 fails to build with gcc 15

gcc 15 is now in Fedora rawhide.  wgrib2 fails to build with it by default due to the change to the C23 standard.  Errors include:
```
[ 21%] Building C object wgrib2/CMakeFiles/wgrib2_lib.dir/fnlist.c.o
cd /builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/redhat-linux-build/wgrib2 && /usr/bin/gcc -Dwgrib2_lib_EXPORTS -I/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/redhat-linux-build -I/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/gctpc/source/include -g -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DGFORTRAN -fPIC -fopenmp -MD -MT wgrib2/CMakeFiles/wgrib2_lib.dir/fnlist.c.o -MF CMakeFiles/wgrib2_lib.dir/fnlist.c.o.d -o CMakeFiles/wgrib2_lib.dir/fnlist.c.o -c /builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:9:12: error: initialization of â€˜int (*)(void)â€™ from incompatible pointer type â€˜int (*)(int,  unsigned char **, float *, unsigned int,  char *, void **)â€™ [-Wincompatible-pointer-types]
    9 |    {"else",f_else, Else, 0, "else, -if ... -else ... -endif", 100},
      |            ^~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:9:12: note: (near initialization for â€˜functions[0].fnâ€™)
In file included from /builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:4:
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.h:11:5: note: â€˜f_elseâ€™ declared here
   11 | int f_else(int mode, unsigned char **sec, float *data, unsigned int ndata, char *inv, void **local);
      |     ^~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:10:14: error: initialization of â€˜int (*)(void)â€™ from incompatible pointer type â€˜int (*)(int,  unsigned char **, float *, unsigned int,  char *, void **, const char *)â€™ [-Wincompatible-pointer-types]
   10 |    {"elseif",f_if, Elseif, 1, "elseif X (POSIX regular expression) conditional on match, -if ... -elseif ... -endif", 200},
      |              ^~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:10:14: note: (near initialization for â€˜functions[1].fnâ€™)
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.h:18:5: note: â€˜f_ifâ€™ declared here
   18 | int f_if(int mode, unsigned char **sec, float *data, unsigned int ndata, char *inv, void **local, const char *arg1);
      |     ^~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:11:17: error: initialization of â€˜int (*)(void)â€™ from incompatible pointer type â€˜int (*)(int,  unsigned char **, float *, unsigned int,  char *, void **, const char *)â€™ [-Wincompatible-pointer-types]
   11 |    {"elseif_fs",f_if_fs, Elseif, 1, "elseif X (fixed string) conditional execution", 200},
      |                 ^~~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:11:17: note: (near initialization for â€˜functions[2].fnâ€™)
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.h:20:5: note: â€˜f_if_fsâ€™ declared here
   20 | int f_if_fs(int mode, unsigned char **sec, float *data, unsigned int ndata, char *inv, void **local, const char *arg1);
      |     ^~~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:12:16: error: initialization of â€˜int (*)(void)â€™ from incompatible pointer type â€˜int (*)(int,  unsigned char **, float *, unsigned int,  char *, void **, const char *)â€™ [-Wincompatible-pointer-types]
   12 |    {"elseif_n",f_if_n, Elseif, 1, "elseif (inv numbers in range), X=(start:end:step)", 200},
      |                ^~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:12:16: note: (near initialization for â€˜functions[3].fnâ€™)
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.h:21:5: note: â€˜f_if_nâ€™ declared here
   21 | int f_if_n(int mode, unsigned char **sec, float *data, unsigned int ndata, char *inv, void **local, const char *arg1);
      |     ^~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:13:18: error: initialization of â€˜int (*)(void)â€™ from incompatible pointer type â€˜int (*)(int,  unsigned char **, float *, unsigned int,  char *, void **, const char *)â€™ [-Wincompatible-pointer-types]
   13 |    {"elseif_rec",f_if_rec, Elseif, 1, "elseif (record numbers in range), X=(start:end:step)", 200},
      |                  ^~~~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:13:18: note: (near initialization for â€˜functions[4].fnâ€™)
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.h:22:5: note: â€˜f_if_recâ€™ declared here
   22 | int f_if_rec(int mode, unsigned char **sec, float *data, unsigned int ndata, char *inv, void **local, const char *arg1);
      |     ^~~~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:14:18: error: initialization of â€˜int (*)(void)â€™ from incompatible pointer type â€˜int (*)(int,  unsigned char **, float *, unsigned int,  char *, void **, const char *)â€™ [-Wincompatible-pointer-types]
   14 |    {"elseif_reg",f_if_reg, Elseif, 1, "elseif rpn registers defined, X = A, A:B, A:B:C, etc A = register number", 200},
      |                  ^~~~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:14:18: note: (near initialization for â€˜functions[5].fnâ€™)
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.h:23:5: note: â€˜f_if_regâ€™ declared here
   23 | int f_if_reg(int mode, unsigned char **sec, float *data, unsigned int ndata, char *inv, void **local, const char *arg1);
      |     ^~~~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:15:13: error: initialization of â€˜int (*)(void)â€™ from incompatible pointer type â€˜int (*)(int,  unsigned char **, float *, unsigned int,  char *, void **)â€™ [-Wincompatible-pointer-types]
   15 |    {"endif",f_endif, Endif, 0, "terminates if block", 100},
      |             ^~~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:15:13: note: (near initialization for â€˜functions[6].fnâ€™)
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.h:17:5: note: â€˜f_endifâ€™ declared here
   17 | int f_endif(int mode, unsigned char **sec, float *data, unsigned int ndata, char *inv, void **local);
      |     ^~~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:16:10: error: initialization of â€˜int (*)(void)â€™ from incompatible pointer type â€˜int (*)(int,  unsigned char **, float *, unsigned int,  char *, void **, const char *)â€™ [-Wincompatible-pointer-types]
   16 |    {"if",f_if, If, 1, "if X (POSIX regular expression), conditional execution on match", 100},
      |          ^~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:16:10: note: (near initialization for â€˜functions[7].fnâ€™)
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.h:18:5: note: â€˜f_ifâ€™ declared here
   18 | int f_if(int mode, unsigned char **sec, float *data, unsigned int ndata, char *inv, void **local, const char *arg1);
      |     ^~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:17:24: error: initialization of â€˜int (*)(void)â€™ from incompatible pointer type â€˜int (*)(int,  unsigned char **, float *, unsigned int,  char *, void **)â€™ [-Wincompatible-pointer-types]
   17 |    {"if_delayed_error",f_if_delayed_error, If, 0, "if delayed error", 100},
      |                        ^~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:17:24: note: (near initialization for â€˜functions[8].fnâ€™)
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.h:19:5: note: â€˜f_if_delayed_errorâ€™ declared here
   19 | int f_if_delayed_error(int mode, unsigned char **sec, float *data, unsigned int ndata, char *inv, void **local);
      |     ^~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/wgrib2-3.5.0-build/wgrib2-3.5.0/wgrib2/fnlist.c:18:13: error: initialization of â€˜int (*)(void)â€™ from incompatible pointer type â€˜int (*)(int,  unsigned char **, float *, unsigned int,  char *, void **, const char *)â€™ [-Wincompatible-pointer-types]
   18 |    {"if_fs",f_if_fs, If, 1, "if X (fixed string), conditional execution on match", 100},
      |             ^~~~~~~
```
this is just a few but I think you get the gist.  See https://gcc.gnu.org/gcc-15/porting_to.html#c23 for some more info.

--- a/wgrib2/gctpc/source/for_init.c
+++ b/wgrib2/gctpc/source/for_init.c
@@ -22,7 +22,7 @@ ALGORITHM REFERENCES
 #include "cproj.h"
 
 void for_init(long outsys, long outzone, double *outparm, long outspheroid,
-        char *fn27, char *fn83, long *iflg, long (*for_trans[])()) {
+        char *fn27, char *fn83, long *iflg, long (*for_trans[])(double, double, double *, double *)) {
 
 //  void for_init(outsys,outzone,outparm,outspheroid,fn27,fn83,iflg,for_trans)
 //  

--- a/wgrib2/gctpc/source/include/proj.h
+++ b/wgrib2/gctpc/source/include/proj.h
@@ -170,7 +170,7 @@ long equiinvint(double r_maj, double center_lon, double lat1,
 	double false_east, double false_north);
 long equiinv(double x, double y, double *lon, double *lat);
 void for_init(long outsys, long outzone, double *outparm, long outspheroid,
-	char *fn27, char *fn83, long *iflg, long (*for_trans[])());
+	char *fn27, char *fn83, long *iflg, long (*for_trans[])(double, double, double *, double *));
 void gctp(double *incoor, long *insys, long *inzone, double *inparm,
 	long *inunit, long *inspheroid, long *ipr, char *efile, long *jpr,
 	char *pfile, double *outcoor, long *outsys, long *outzone,
@@ -203,7 +203,7 @@ long imolwfor(double lon, double lat, double *x, double *y);
 long imolwinvint(double r);
 long imolwinv(double x, double y, double *lon, double *lat);
 void inv_init(long insys, long inzone, double *inparm, long inspheroid,
-	char *fn27, char *fn83, long *iflg, long (*inv_trans[])());
+	char *fn27, char *fn83, long *iflg, long (*inv_trans[])(double, double, double *, double *));
 long lamazforint(double r, double center_long, double center_lat,
 	double false_east, double false_north);
 long lamazfor(double lon, double lat, double *x, double *y);

--- a/wgrib2/gctpc/source/inv_init.c
+++ b/wgrib2/gctpc/source/inv_init.c
@@ -22,7 +22,7 @@ ALGORITHM REFERENCES
 #include "cproj.h"
 
 void inv_init(long insys, long inzone, double *inparm, long inspheroid,
-        char *fn27, char *fn83, long *iflg, long (*inv_trans[])()) {
+        char *fn27, char *fn83, long *iflg, long (*inv_trans[])(double, double, double *, double *)) {
 //void inv_init(insys,inzone,inparm,inspheroid,fn27,fn83,iflg,inv_trans)
 //
 //long insys;		/* input system code				*/

